libraries/rush-lib/src/logic/operations/test/BuildPlanPlugin.test.ts

```diff
@@ -0,0 +1,133 @@
+  });
+
+  function createMockRunner(operations: Set<Operation>, context: ICreateOperationsContext): Set<Operation> {
+    for (const operation of operations) {
+      const { associatedPhase, associatedProject } = operation;
+
+      if (associatedPhase && associatedProject && !operation.runner) {
+        const name: string = `${associatedProject.packageName} (${associatedPhase.name.slice(
+          RushConstants.phaseNamePrefix.length
+        )})`;
+
+        operation.runner = new MockOperationRunner(name, undefined, undefined, false);
+      }
+    }
+
+    return operations;
+  }
+
+  async function testCreateOperationsAsync(
+    phaseSelection: Set<IPhase>,
+    projectSelection: Set<RushConfigurationProject>,
+    changedProjects: Set<RushConfigurationProject>
+  ): Promise<Set<Operation>> {
+    const hooks: PhasedCommandHooks = new PhasedCommandHooks();
+    // Apply the plugin being tested
+    new PhasedOperationPlugin().apply(hooks);
+    // Add mock runners for included operations.
+    hooks.createOperations.tap('MockOperationRunnerPlugin', createMockRunner);
+
+    const context: Pick<
+      ICreateOperationsContext,
+      'phaseOriginal' | 'phaseSelection' | 'projectSelection' | 'projectsInUnknownState'
+    > = {
+      phaseOriginal: phaseSelection,
+      phaseSelection,
+      projectSelection,
+      projectsInUnknownState: changedProjects
+    };
+    const operations: Set<Operation> = await hooks.createOperations.promise(
+      new Set(),
+      context as ICreateOperationsContext
+    );
+
+    return operations;
+  }
+
+  describe('build plan debugging', () => {
+    it('should generate a build plan', async () => {
+      const hooks: PhasedCommandHooks = new PhasedCommandHooks();
+
+      new BuildPlanPlugin(terminal).apply(hooks);
+      const context: Pick<IExecuteOperationsContext, 'projectChangeAnalyzer' | 'projectConfigurations'> = {
+        projectChangeAnalyzer: {
+          [ProjectChangeAnalyzer.prototype._tryGetProjectDependenciesAsync.name]: async () => {
+            return new Map();
+          }
+        } as unknown as ProjectChangeAnalyzer,
+        projectConfigurations: new Map()
+      };
+      const buildCommand: IPhasedCommandConfig = commandLineConfiguration.commands.get(
+        'build'
+      )! as IPhasedCommandConfig;
+
+      const operationMap = new Map();
+
+      const operations = await testCreateOperationsAsync(
+        buildCommand.phases,
+        new Set(rushConfiguration.projects),
+        new Set(rushConfiguration.projects)
+      );
+      operations.forEach((operation) => {
+        operationMap.set(
+          operation,
+          new OperationExecutionRecord(operation, { debugMode: false, quietMode: true, streamCollator })
+        );
+      });
+
+      await hooks.beforeExecuteOperations.promise(operationMap, context as IExecuteOperationsContext);
+
+      expect(stringBufferTerminalProvider.getOutput()).toMatchSnapshot();
```

Comment:
```suggestion
      expect(stringBufferTerminalProvider.getOutput({ normalizeSpecialCharacters: false })).toMatchSnapshot();
```

To get rid of the `[n]`s.